pmlc/util/ids.h

```diff
@@ -0,0 +1,10 @@
+// Copyright 2020 Intel Corporation
+#pragma once
+
+// A collection of well-known identifiers used in codegen.
```

Comment:
Do we really need a whole separate file just for this?

I think this probably belongs in `compiler/program.h`

Reply:
Files are cheap, and `JitExecutable` will need access to these identifiers; it'd be nice to not take a dependency on `compiler`.

Reply:
JitExecutable should depend on compiler shouldn't it? We need the `Program` to be passed along, that's where the constant buffers live.

Reply:
JitExecutable currently depends on compiler, but IMHO that's an antipattern: `Program::compile` isn't an intrinsic part of a `Program` object, it's a member of Program because it has to do with programs and humans like categorizing things; it could be easily be a free function.  (I do this all the time myself, which is why I'm on somewhat guard against it.  :-)

I also don't think these identifiers are necessarily part of Program -- unless we want to constrain Program to always providing these particular interfaces?  But I think of them as a handshake between a particular target and JitExecutable.

Reply:
The constant buffers lifetime are tied to the `Program`. We need to pass the `program` to an `executable` so that the program can be executed. I don't see the antipattern here.

Reply:
The antipattern is that `Program::compile()` exists.  It would be better for `Program` to be moved somewhere else, and to think of compilation as a transformation applied to a `Program` datastructure instead of something that's intrinsic to the definition of a `Program`.

(Or is there another reason why `Program` is defined under `pmlc/compiler`?)

Reply:
I still don't see the issue.